src/Bindings/ManualBindings.cpp

```diff
@@ -239,21 +244,19 @@ static int tolua_CompressStringGZIP(lua_State * tolua_S)
 	}
 
 	// Get the params:
-	AString ToCompress;
+	ContiguousByteBuffer ToCompress;
 	S.GetStackValues(1, ToCompress);
 
 	// Compress the string:
-	AString res;
-	CompressStringGZIP(ToCompress.data(), ToCompress.size(), res);
-	S.Push(res);
+	S.Push(Compression::Compressor().CompressGZip(ToCompress).GetView());
```

Comment:
may throw

Reply:
actually only if memory allocation fails. That should probably fall under a general strategy of memory failures in the API surface/internal code, instead of handling this one specific instance.